app/models/singly_linked_list.rb

```diff
@@ -0,0 +1,77 @@
+class SinglyLinkedList
+  include Enumerable
```

Comment:
I agree that we shouldn't test `#grep`, `#include?`, etc. But the `Enumerable` module is basically designed to work correctly as long as there is an `#each` method defined. From http://ruby-doc.org/core-2.3.0/Enumerable.html:

> The Enumerable mixin provides collection classes with several traversal and searching methods, and with the ability to sort. The class must provide a method each, which yields successive members of the collection.

I think we can trust Enumerable as long as `each` yields each member of the list in turn, it's pretty robust. In fact, it even works with an infinite `each` method. For example:

```ruby
class Fib
  include Enumerable

  def each
    first, second = 0, 1
    loop do
      yield second
      first, second = second, first + second
    end
  end
end

Fib.new.first(10) # => [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
```